### PR TITLE
Ensure `ArmeriaServerConfigurator` overrides properties from `ArmeriaSettings`

### LINKED ISSUE #5009: Programmatically configured options on `ServerBuilder` are overridden by `ArmeriaSettings`
When using the `spring-boot{2,3}-autoconfigure` library, the provided `ArmeriaSettings` autoconfiguration is overridding any user-supplied configuration in code.

For example, when using `builder.gracefulShutdownTimeout(x, y)`, the provided values are applied to`ServerBuilder`, but before the server is built, they're overwritten by the autoconfiguration classes.

This can be reproduced by setting a breakpoint here:
https://github.com/line/armeria/blob/2e506361b64356d04dcae0eebce155dc8adf3743/core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java#L824-L825

Then, you should set a custom value here:
https://github.com/line/armeria/blob/2e506361b64356d04dcae0eebce155dc8adf3743/examples/spring-boot-minimal/src/main/java/example/springframework/boot/minimal/HelloConfiguration.java#L21-L24

Once the application is started, you can observe the provided value being overriden by `ArmeriaSettings`.

As a temporary workaround, you can inject `ArmeriaSettings` and use `setGracefulShutdownQuietPeriodMillis()`

### PR DESCRIPTION
**Pull request by minwoox**
Motivation:
The properties from `ArmeriaSettings` should be considered as default values and should be overridden by the beans that implement `ArmeriaServerConfigurator`. This ensures that custom configurations provided by `ArmeriaServerConfigurator` take precedence over the default settings.

Modifications:
- Adjusted the order of configuration application so that `ArmeriaServerConfigurator` is applied last, ensuring it can override properties set by `ArmeriaSettings`.

Result:
- `ArmeriaServerConfigurator` now properly overrides the properties set by `ArmeriaSettings`.
- Close #5009

### CODE REVIEW

**Review by KarboniteKream — APPROVED** [review]
Thank you for addressing this!

File: spring/boot3-autoconfigure/src/test/resources/config/application-settings.yml [hunk 1]
```diff
@@ -17,3 +17,5 @@ armeria:
   request-timeout: 8000
   max-request-length: 0
   verbose-responses: true
+  graceful-shutdown-quiet-period-millis: 2000
+  graceful-shutdown-timeout-millis: 20000
```
Comment (jrhee17) [inline spring/boot3-autoconfigure/src/test/resources/config/application-settings.yml:21]: nit;

```suggestion
    graceful-shutdown-timeout-millis: 20000
  
```

**Review by jrhee17 — APPROVED** [review]
👍 👍 👍

File: spring/boot3-autoconfigure/src/test/resources/config/application-settings.yml (hunk 1 above)
Comment (minwoox) [reply spring/boot3-autoconfigure/src/test/resources/config/application-settings.yml]: Oops thanks!

**Comment by minwoox** [comment]
Failed tests will be handled by https://github.com/line/armeria/pull/5694

**Review by ikhoon — APPROVED** [review]

**Review by trustin — APPROVED** [review]
Thanks, @minwoox !

**Pull request merged**